### PR TITLE
Improve the Exception handling for threads by using queues. Fix T2031

### PR DESCRIPTION
**Pull request by sanacl**
Previous implementation was not raising correctly the exceptions, due to there was not method to control the exceptions launched from the mother/father process. 

How to reproduce the error?
- included this mbox data source in .json file
```
	    "mbox":[
	    	"/home/luis/repos/mordred/mboxes"
	    ]
```
- executed with update mode
- monitored the number of threads
```
watch -n1 "ps huH p `ps aux|grep mordred|grep python|awk {'print $2'}`"
```
- while the error is printed in the terminal were mordred was executed, the error is not logged and one of the threads dies

Current changes allow threads to share exceptions using a queue, so the main process can wait and read if any exception was raised and needs to be addressed.

The main process does not wait and check always, in this case the exception is raised from the thread but the traceback is not printed, as this is something done by the parent. This is something to be improved.

### CODE REVIEW

File: mordred/task_manager.py [hunk 1]
```diff
@@ -84,7 +86,9 @@ def run(self):
                 time.sleep(self.timer)
 
             for task in self.tasks:
-                task.run()
+                try:
+                    task.run()
```
Comment (acs) [inline mordred/task_manager.py]: Ups, you can not start threads in this way. You must use task.start() which really create a new thread and then execute "run" with it. If you call directly to run, you are not creating a new thread, but just executing the run method in the thread object. This is why you can catch the Exception in this point.

If you use "start" a new thread is started and, if there are exceptions in the thread, you won't catch them here. You need to catch the exception inside the thread and put it in the queue, but inside the thread.

**Comment by sanacl** [comment]
You are getting confused with the names. 

The task.run() you see is a call received by an object which inherits from Task(), not from a Thread class.

The class which inherits from Threading is TasksManager.

The Task Class has a run method like this
```
    def run(self):
        """ Execute the Task """
        logger.debug("A bored task. It does nothing!")
```

The threads are being executed calling to the start() method in methods like execute_batch_tasks() (in mordred.py)

**Comment by sanacl** [comment]
I'm updating the code now so the merge with master is painless.

**Comment by acs** [comment]
Ok, all done! Good job @sanacl Time to enjoy the new feature!

**Comment by sanacl** [comment]
@acs I'm glad you like it. I'm doing my best trying to remember & improve my python skills. Any feedback about improving it is welcome.

One question, what do you guys think we should do with this branches? Should we remove them?

**Pull request merged**